src/classes/RD2_ApiService.cls

```diff
@@ -55,7 +55,7 @@ public inherited sharing class RD2_ApiService {
 
             List<RD2_ApiService.Schedule> schedules = new List<RD2_ApiService.Schedule>();
             for (RecurringDonationSchedule__c s : rd.RecurringDonationSchedules__r) {
-                schedules.add(new RD2_ApiService.Schedule(s, rd));
+                schedules.add(new RD2_ApiService.Schedule(s, rd, getCurrencyCode(rd)));
```

Comment:
Since the RD is passed into the `RD2_ApiService.Schedule`, then `getCurrencyCode(rd)` should be the schedule responsibility, ie part of that class. Or make the method static so it can be accessed from `Schedule` and `Installment` inner classes.
Suggestion: the `RD2_ApiService.Schedule` inner class can be called `Schedule` unless there are compile issues.